lib/engine.js

```diff
@@ -350,81 +347,76 @@
     return JSON.stringify(state);
   }
 
-  let initialRNG = new JKISS31();
-  initialRNG.scramble();
-  initialRNG = initialRNG.serialize();
-
-  let initialState = JSON.stringify({
-    time: 0,
-    flashTime: 3,
-    floatTime: 2,
-    swapTime: 2,
-    chainNumber: 0,
-    initialRows: INITIAL_ROWS,
-    RNG: initialRNG,
-    nextRow: null,
-    blocks: (() => {
-      const blocks = [];
-      for (let i = 0; i < WIDTH * HEIGHT; ++i) {
-        blocks.push(newBlock());
-      }
-      return blocks;
-    })(),
-  });
-
-  return {
-    width: WIDTH,
-    height: HEIGHT,
-
-    getTime() {
-      return time;
-    },
+  function GameEngine() {
+    const initialRNG = new JKISS31();
+
+    this.width = WIDTH;
+    this.height = HEIGHT;
+    this.time = 0;
+    this.eventQueue = [];
+    initialRNG.scramble();
+    this.initialState = JSON.stringify({
+      time: 0,
+      flashTime: 3,
+      floatTime: 2,
+      swapTime: 2,
+      chainNumber: 0,
+      initialRows: INITIAL_ROWS,
+      RNG: initialRNG.serialize(),
+      nextRow: null,
+      blocks: (() => {
+        const blocks = [];
+        for (let i = 0; i < WIDTH * HEIGHT; ++i) {
+          blocks.push(newBlock());
+        }
+        return blocks;
+      })(),
+    });
+  }
 
-    setTime(newTime) {
-      time = newTime;
-    },
+  GameEngine.width = WIDTH;
```

Comment:
You're not using these class level constants anywhere. Also width and height will be instance specific in the future so adding them at class level doesn't make any sense.